codec-http/src/test/java/io/netty/handler/codec/http/HttpServerExpectContinueHandlerTest.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2017 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.codec.http;
+
+import io.netty.channel.embedded.EmbeddedChannel;
+import org.junit.Test;
+
+import static org.hamcrest.CoreMatchers.*;
+import static org.junit.Assert.*;
+
+public class HttpServerExpectContinueHandlerTest {
+
+    @Test
+    public void test_RespondToExpectHeader() {
+        EmbeddedChannel channel = new EmbeddedChannel(new HttpServerExpectContinueHandler() {
+            @Override
+            protected HttpResponse acceptMessage(HttpRequest request) {
+                HttpResponse response = new DefaultFullHttpResponse(HttpVersion.HTTP_1_1, HttpResponseStatus.CONTINUE);
+                response.headers().set("foo", "bar");
+                return response;
+            }
+        });
+        HttpRequest request = new DefaultFullHttpRequest(HttpVersion.HTTP_1_1, HttpMethod.GET, "/");
+        HttpUtil.set100ContinueExpected(request, true);
+
+        channel.writeInbound(request);
+        HttpResponse response = channel.readOutbound();
+
+        assertThat(response.status(), is(HttpResponseStatus.CONTINUE));
+        assertThat(response.headers().get("foo"), is("bar"));
+
```

Comment:
add a call to `ReferenceCountUtil.release(response);` after you are done using it.